packages/tests/tests/filter.node.spec.ts

```diff
@@ -115,3 +115,107 @@ describe("Waku Filter", () => {
     expect(messageCount).to.eq(1);
   });
 });
+
+describe("Waku Filter: V2", () => {
+  let waku: LightNode;
+  let nwaku: Nwaku;
+
+  afterEach(async function () {
+    !!nwaku &&
+      nwaku.stop().catch((e) => console.log("Nwaku failed to stop", e));
+    !!waku && waku.stop().catch((e) => console.log("Waku failed to stop", e));
+  });
+
+  beforeEach(async function () {
+    this.timeout(15000);
+    nwaku = new Nwaku(makeLogFileName(this));
+    // New Filter protocol only works on go-waku right now
+    if (nwaku.nodeType !== "go-waku") this.skip();
+    await nwaku.start({
+      filter: true,
+      lightpush: true,
+      relay: true,
+      useFilterv2: true,
+    });
+    waku = await createLightNode({
+      useFilterV2: true,
+      staticNoiseKey: NOISE_KEY_1,
+      libp2p: { addresses: { listen: ["/ip4/0.0.0.0/tcp/0/ws"] } },
+    });
+    await waku.start();
+    await waku.dial(await nwaku.getMultiaddrWithId());
+    await waitForRemotePeer(waku, [Protocols.Filter, Protocols.LightPush]);
+  });
+
+  it("creates a subscription", async function () {
```

Comment:
Need to also test the new features: `PING`, `UNSUBSCRIBE_ALL` and using `SUBSCRIBE` to modify a subscription.